source/Engine.cpp

```diff
@@ -870,9 +866,29 @@ void Engine::Step(bool isActive)
 	// Draw crosshairs on any minables in range of the flagship's scanners.
 	double scanRange = flagship ? 100. * sqrt(flagship->Attributes().Get("asteroid scan power")) : 0.;
 	if(flagship && scanRange && !flagship->IsHyperspacing())
+	{
+		// Decide before looping whether or not to catalog asteroids.  This
+		// results in cataloging in-range asteroids roughly 3 times a second.
+		bool shouldCatalogAsteroids = !Random::Int(20);
+		shouldCatalogAsteroids &= !isAsteroidCatalogComplete;
```

Comment:
```suggestion
		bool shouldCatalogAsteroids = (!isAsteroidCatalogComplete && !Random::Int(20));
```
This way we don't calculate the random when we don't have to.
Parenthesis isnt required but I like it ^^